src/main/java/org/broadinstitute/hellbender/tools/spark/sv/AlignmentRegion.java

```diff
@@ -0,0 +1,169 @@
+
+    final String contigId;
+    final String breakpointId;
+    final Cigar cigar;
+    final boolean forwardStrand;
+    final SimpleInterval referenceInterval;
+    final int mqual;
+    final int startInAssembledContig;
+    final int endInAssembledContig;
+    final int assembledContigLength;
+    final int mismatches;
+
+    public AlignmentRegion(final String breakpointId, final String contigId, final AlnRgn alnRgn) {
+        this.contigId = contigId;
+        this.breakpointId = breakpointId;
+        this.forwardStrand = alnRgn.getStrand() == '+';
+        final Cigar alignmentCigar = TextCigarCodec.decode(alnRgn.getCigar());
+        this.cigar = forwardStrand ? alignmentCigar : CigarUtils.invertCigar(alignmentCigar);
+        this.referenceInterval = new SimpleInterval(alnRgn.getChrom(), (int) alnRgn.getPos() + 1, (int) (alnRgn.getPos() + 1 + cigar.getReferenceLength()));
+        this.mqual = alnRgn.getMQual();
+        this.assembledContigLength = cigar.getReadLength();
+        this.startInAssembledContig = startOfAlignmentInContig(cigar);
+        this.endInAssembledContig = endOfAlignmentInContig(assembledContigLength, cigar);
+        this.mismatches = alnRgn.getNm();
+    }
+
+    public AlignmentRegion(final String breakpointId, final String contigId, final Cigar cigar, final boolean forwardStrand, final SimpleInterval referenceInterval, final int mqual, final int startInAssembledContig, final int endInAssembledContig, final int mismatches) {
+        this.contigId = contigId;
+        this.breakpointId = breakpointId;
+        this.cigar = cigar;
+        this.forwardStrand = forwardStrand;
+        this.referenceInterval = referenceInterval;
+        this.mqual = mqual;
+        this.startInAssembledContig = startInAssembledContig;
+        this.endInAssembledContig = endInAssembledContig;
+        this.assembledContigLength = cigar.getReadLength();
+        this.mismatches = mismatches;
+    }
+
+    public AlignmentRegion(final GATKRead read) {
+        this.breakpointId = null;
+        this.contigId = read.getName();
+        this.forwardStrand = ! read.isReverseStrand();
+        this.cigar = forwardStrand ? read.getCigar() : CigarUtils.invertCigar(read.getCigar());
+        this.referenceInterval = new SimpleInterval(read);
+        this.assembledContigLength = cigar.getReadLength() + getHardClipping(cigar);
+        this.startInAssembledContig = startOfAlignmentInContig(cigar);
+        this.endInAssembledContig = endOfAlignmentInContig(assembledContigLength, cigar);
+        this.mqual = read.getMappingQuality();
+        if (read.hasAttribute("NM")) {
+            this.mismatches = read.getAttributeAsInteger("NM");
+        } else {
+            this.mismatches = 0;
+        }
+    }
+
+    public int overlapOnContig(final AlignmentRegion other) {
+        return Math.max(0, Math.min(endInAssembledContig, other.endInAssembledContig) - Math.max(startInAssembledContig, other.startInAssembledContig));
+    }
```

Comment:
should there be a "+1"? Take example of two contigs A[10,100], B[80, 120], should the overlap be 100-80+1? or the interval is right-exclusive?